src/auth/__tests__/PrivateRoute.unit.test.tsx

```diff
@@ -0,0 +1,203 @@
+        </Switch>
+      </Router>,
+      { wrapper: AuthProvider }
+    );
+
+    act(() => {
+      mockAxios.mockResponse({ data: { success: true, data: testUser } });
+    });
+
+    expect(getByText("404")).toBeInTheDocument();
+  });
+
+  it("displays content for logged-in users", () => {
+    const { getByText } = render(
+      <Router history={history}>
+        <Switch>
+          <PrivateRoute path="/private">
+            Secret
+          </PrivateRoute>
+        </Switch>
+      </Router>,
+      { wrapper: AuthProvider }
+    );
+
+    act(() => {
+      mockAxios.mockResponse({ data: { success: true, data: testUser } });
+    });
+
+    expect(getByText("Secret")).toBeInTheDocument();
+  });
+
+  it("displays admin content for admins", () => {
+    const { getByText } = render(
+      <Router history={history}>
+        <Switch>
+          <PrivateRoute admin path="/private">
+            Secret
+          </PrivateRoute>
+        </Switch>
+      </Router>,
+      { wrapper: AuthProvider }
+    );
+
+    act(() => {
+      mockAxios.mockResponse({ data: { success: true, data: testAdmin } });
+    });
+
+    expect(getByText("Secret")).toBeInTheDocument();
+  });
+
+  it("shows error page when admin has been downgraded to user", () => {
+    localStorage.setItem(USER_LOCALSTORAGE_KEY, JSON.stringify(testAdmin));
+
+    const { getByText } = render(
+      <Router history={history}>
+        <Switch>
+          <PrivateRoute admin path="/private">
+            Secret
+          </PrivateRoute>
+        </Switch>
+      </Router>,
+      { wrapper: AuthProvider }
+    );
+
+    act(() => {
+      mockAxios.mockResponse({ data: { success: true, data: testUser } });
+    });
+
+    expect(getByText("404")).toBeInTheDocument();
+  });
+
+  it("shows error page when user has been logged out elsewhere", () => {
+    localStorage.setItem(USER_LOCALSTORAGE_KEY, JSON.stringify(testUser));
+
+    const { getByText } = render(
+      <Router history={history}>
+        <Switch>
+          <PrivateRoute path="/private">
+            Secret
+          </PrivateRoute>
+        </Switch>
+      </Router>,
+      { wrapper: AuthProvider }
+    );
+
+    act(() => {
+      mockAxios.mockResponse({ data: { success: true, data: null } });
+    });
+
+    expect(getByText("404")).toBeInTheDocument();
+  });
+
+  it("shows errors when auth check fails", () => {
+    const { getByText } = render(
+      <Router history={history}>
+        <Switch>
+          <PrivateRoute path="/private">
+            Secret
+          </PrivateRoute>
+        </Switch>
+      </Router>,
+      { wrapper: AuthProvider }
+    );
+
+    act(() => {
+      mockAxios.mockError({
+        code: 500,
+        response: {
+          data: {
+            success: false,
+            error: {
+              detail: ["Error thrown in testing"]
+            }
+          }
+        }
+      });
+    });
+
+    expect(getByText("Error thrown in testing")).toBeInTheDocument();
```

Comment:
I recognize we're doing this already, but do we want to be directly outputting auth errors into the document? We should run them through the errors map first.

Reply:
I believe they're already being run through the errors map. This is just testing that the specific error message returned is being displayed.

Reply:
Unless I'm missing something, we're just outputting them directly:

```
return <>
  {state.errors.map(err => <Alert key={err} variant="danger">{err}</Alert>)}
</>;
```

Reply:
Oh, I see what you mean. Yeah, that's a good call; I'll also test to make sure errors are being passed through the map on all of the tests.

Reply:
52141354